### PR TITLE
add headers for version

### LINKED ISSUE #7926: [TypeScript SDK] Add headers for version
Rust SDK has added two headers headers in https://github.com/MystenLabs/sui/blob/faa880cde9393aa9cceadbd5b3582be81c7997e9/crates/sui-sdk/src/lib.rs#L78-L82 to help us analyze the distribution of SDK versions. This issue is to add these two headers in TS SDK as well: https://github.com/MystenLabs/sui/blob/main/sdk/typescript/src/rpc/client.ts#L58

### PR DESCRIPTION
**Pull request by danaiballa**
Closes #7926

### CODE REVIEW

**Review by teohaik — CHANGES_REQUESTED** [review]

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -547,6 +548,7 @@ importers:
       eslint-config-prettier: 8.6.0_eslint@8.31.0
       eslint-config-react-app: 7.0.1_iukboom6ndih5an6iafl45j2fe
       eslint-plugin-prettier: 4.2.1_32m5uc2milwdw3tnkcq5del26y
+      genversion: 3.1.1
       husky: 7.0.4
```
[the remark this answers is not in the document]
Comment (danaiballa) [reply pnpm-lock.yaml:552]: @teohaik Are you sure that this is an issue?  I think I installed genversion only once and this happened automatically. This is also happening to a lot of packages in this file, being mentioned 2 times, once under `specifiers` and once under either `dependencies` or `devDependencies`, for example `husky`, 'jayson', 'eslint-plugin-prettier`.

File: crates/sui-json-rpc/src/lib.rs [hunk 2]
```diff
@@ -86,12 +87,15 @@ impl JsonRpcServerBuilder {
         };
         info!(?acl);
 
+        let client_type: HeaderName = HeaderName::from_static("client-type");
+        let client_api_version: HeaderName = HeaderName::from_static("client-api-version");
+
```
Comment (666lcz) [inline crates/sui-json-rpc/src/lib.rs:92]: @patrickkuo do you know if this is necessary?

File: .github/workflows/changesets.yml [hunk 3]
```diff
@@ -47,5 +47,9 @@ jobs:
         run: pnpm install --frozen-lockfile
       - name: Create Release Pull Request
         uses: changesets/action@e9cc34b540dd3ad1b030c57fd97269e8f6ad905a # pin@v1
+        with:
+          # Generates src/version.ts file using genVersion
+          # https://github.com/changesets/action#with-version-script
+          version: pnpm run codegen:version
```
Comment (666lcz) [inline .github/workflows/changesets.yml:53]: @Jordan-Mysten do you know if there's a good way to test this change(e.g. verify that this is included in the new Package Version PR) without merging this PR?

File: crates/sui-json-rpc/src/lib.rs (hunk 2 above)
Comment (patrickkuo) [reply crates/sui-json-rpc/src/lib.rs:91]: yes we probably need this when calling from a different host

**Review by 666lcz — APPROVED** [review]

**Pull request merged**